src/App.tsx

```diff
@@ -1,51 +1,146 @@
-          Jam is selected
-        </h1>
-
-        <button
-          type="button"
-          className="App__clear"
-        >
-          Clear
-        </button>
-      </header>
-
-      <ul>
-        <li className="Good">Dumplings</li>
-        <li className="Good">Carrot</li>
-        <li className="Good">Eggs</li>
-        <li className="Good">Ice cream</li>
-        <li className="Good">Apple</li>
-        <li className="Good">Bread</li>
-        <li className="Good">Fish</li>
-        <li className="Good">Honey</li>
-        <li className="Good Good--active">Jam</li>
-        <li className="Good">Garlic</li>
-      </ul>
-
-      {/* Put required buttons into each Good */}
-      <button
-        type="button"
-        className="Good__remove"
-      >
-        Remove
-      </button>
-
-      <button
-        type="button"
-        className="Good__select"
-      >
-        Select
-      </button>
-    </main>
-  );
+import classNames from 'classnames';
+import goodsFromServer from './goods';
+
+type State = {
+  selectedGood: string[];
+  isSelectedDefault: boolean;
 };
+
+export class App extends Component<{}, State> {
+  state: Readonly<State> = {
+    selectedGood: ['Jam'],
+    isSelectedDefault: true,
+  };
+
+  goods: string[] = [...goodsFromServer];
+
+  isActive = (good: string) => {
+    return !!(this.state.selectedGood.find(item => item === good));
+  };
+
+  removeItem = (good: string) => (
+    this.setState((prevState) => {
+      const { selectedGood } = prevState;
+      const indexFinde = selectedGood
+        .findIndex(item => item === good);
+
+      if (indexFinde !== -1) {
+        selectedGood.splice(indexFinde, 1);
+      }
+
+      return {
+        selectedGood: [...selectedGood],
+      };
+    })
+  );
+
+  clearState = () => {
+    this.setState({
+      selectedGood: [],
+    });
+  };
+
+  setHeader = (): string => {
+    const lengthSelected = this.state.selectedGood.length;
+
+    if (!(lengthSelected)) {
+      return 'No goods selected';
+    }
+
+    if (lengthSelected === 1) {
+      return `${this.state.selectedGood[0]} is selected`;
+    }
+
+    const selectedWithoutLastElement = [...this.state.selectedGood];
+
+    const lastSelected = selectedWithoutLastElement.pop();
+
+    return `${selectedWithoutLastElement.join(', ')} and ${lastSelected} are selected`;
+  };
+
+  addeItem = (good: string): void => {
+    if (this.state.isSelectedDefault) {
+      this.clearState();
+      this.setState(() => {
+        return {
+          isSelectedDefault: false,
+        };
+      });
+    }
+
+    this.setState((prevState) => {
+      const { selectedGood } = prevState;
+
+      return {
+        selectedGood: [...selectedGood, good],
+      };
+    });
+  };
+
+  render() {
+    return (
+      <main className="App">
+        <header className="App__header">
+          <h1 className="App__title">
+            { this.setHeader() }
+          </h1>
+          {!!(this.state.selectedGood.length) && (
```

Comment:
```suggestion
          {this.state.selectedGood.length && (
```

double !! is useless, it will be the same as without them

![image](https://user-images.githubusercontent.com/57747481/183379347-5fa78e9f-e9d9-490e-83a2-869f255b382b.png)